### DISCUSSION TITLE
Different MAIL FROM and DATA's From, and different RCPT TO and DATA's To

### DISCUSSION THREAD

**Discussion by Adaptogen90** [jstedfast/MailKit#1662] [Q&A] [answered]
Is it possible to accomplish the following using Mailkit?
- a MAIL FROM that is different than DATA's From header
- an RCPT TO that is different than DATA's To header
If not, any other possible solutions aside from directly issuing SMTP commands in c#?
Many thanks in advance!

**Comment by jstedfast** [comment, marked as answer]
Yes. Instead of using `SmtpClient.Send(MimeMessage message`), use `SmtpClient.Send(MimeMessage message, MailboxAddress sender, IEnumerable<MailboxAddress> recipients)`

**Comment by Adaptogen90** [comment]
Amazing! I confirm it works as expected! This saved me from considering a
different language altogether that handles this better natively. Many
thanks.
